nano/core_test/node.cpp

```diff
@@ -1682,26 +1682,40 @@ TEST (node, bootstrap_confirm_frontiers)
 }
 
 // Test that if we create a block that isn't confirmed, we sync.
-TEST (node, DISABLED_unconfirmed_send)
+TEST (node, unconfirmed_send)
 {
-	nano::system system (2);
-	auto & node0 (*system.nodes[0]);
-	auto & node1 (*system.nodes[1]);
-	auto wallet0 (system.wallet (0));
-	auto wallet1 (system.wallet (1));
-	nano::keypair key0;
-	wallet1->insert_adhoc (key0.prv);
-	wallet0->insert_adhoc (nano::dev::genesis_key.prv);
-	auto send1 (wallet0->send_action (nano::dev::genesis->account (), key0.pub, 2 * nano::Mxrb_ratio));
-	ASSERT_TIMELY (10s, node1.balance (key0.pub) == 2 * nano::Mxrb_ratio && !node1.bootstrap_initiator.in_progress ());
-	auto latest (node1.latest (key0.pub));
-	nano::state_block send2 (key0.pub, latest, nano::dev::genesis->account (), nano::Mxrb_ratio, nano::dev::genesis->account (), key0.prv, key0.pub, *node0.work_generate_blocking (latest));
-	{
-		auto transaction (node1.store.tx_begin_write ());
-		ASSERT_EQ (nano::process_result::progress, node1.ledger.process (transaction, send2).code);
-	}
-	auto send3 (wallet1->send_action (key0.pub, nano::dev::genesis->account (), nano::Mxrb_ratio));
-	ASSERT_TIMELY (10s, node0.balance (nano::dev::genesis->account ()) == nano::dev::constants.genesis_amount);
+	nano::system system{ 2 };
+
+	auto & node1 = *system.nodes[0];
+	auto wallet1 = system.wallet (0);
+	wallet1->insert_adhoc (nano::dev::genesis_key.prv);
+
+	nano::keypair key2{};
+	auto & node2 = *system.nodes[1];
+	auto wallet2 = system.wallet (1);
+	wallet2->insert_adhoc (key2.prv);
+
+	// firstly, send two units from node1 to node2 and expect that both nodes see the block as confirmed
+	// (node1 will start an election for it, vote on it and node2 gets synced up)
+	auto send1 = wallet1->send_action (nano::dev::genesis->account (), key2.pub, 2 * nano::Mxrb_ratio);
+	ASSERT_TIMELY (5s, node1.block_confirmed (send1->hash ()));
+	ASSERT_TIMELY (5s, node2.block_confirmed (send1->hash ()));
+	ASSERT_TIMELY (10s, node2.balance (key2.pub) == 2 * nano::Mxrb_ratio);
+
+	// then send one of the received units back from node2 to node1;
+	//
+	auto const latest = node2.latest (key2.pub);
+	auto send2 = nano::state_block_builder{}.make_block ().account (key2.pub).previous (latest).representative (nano::dev::genesis_key.pub).balance (nano::Mxrb_ratio).link (nano::dev::genesis->account ()).sign (key2.prv, key2.pub).work (*system.work.generate (latest)).build_shared ();
+	node2.process_active (send2);
```

Comment:
I think that this should not be a process_active. I think this should just insert a block into the ledger bypassing block processing. That is probably what is meant by "unconfirmed send". The original unit test calls `node1.ledger.process`

Reply:
Not sure about that to be honest but I addressed it by using `node.process` now.